lib/bricks-evaluate.js

```diff
@@ -324,17 +361,103 @@ function transformCode(code, options) {
 }
 
 function evaluate(code, context, filename, callback, options) {
-	// Put a reference to `when` into the context.
-	// We generate random identifier to prevent exploiting it from the provided code.
-	var whenIdentifierName = '__when' + Math.floor(1000 + Math.random() * 1000);
-	context[whenIdentifierName] = when;
-	
 	var timing;
 	try {
+		var SPACE_RE = /\s+/g;
+		var TRIM_SPACE_RE = /(^\s+)|(\s+$)/g;
+	
+		// Matches "[ &anything ] ( Type123 name_123 ) {"
+		var CPP_LAMBDA_RE = /(\[([^\]]*?)\]\s*\(([^)]*?)\))\s*\{/g;
+	
+		// Matches "name_123"
+		var CPP_IDENTIFIER_RE = /^([a-zA-Z_][a-zA-Z0-9_]*)\s*$/;
```

Comment:
Fixed the comment in #10. The regexp actually matches anything between `[` and `]` and between the following `(` and `)`.